sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -378,10 +378,9 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
     )
   }
 
-  def dataTypeOperationUnsupportedError(): SparkUnsupportedOperationException = {
-    new SparkUnsupportedOperationException(
-      errorClass = "_LEGACY_ERROR_TEMP_2009",
-      messageParameters = Map.empty)
+  def dataTypeOperationUnsupportedError(): Throwable = {
+    SparkException.internalError(
+      s"""Operation dataType is not supported""")
```

Comment:
```suggestion
      "The operation `dataType` is not supported.")
```

Reply:
Could you apply this, please.

Reply:
Oh... didn't recognize that it's overwritten.

Just applied the change.